src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs

```diff
@@ -433,25 +429,25 @@ protected virtual ParameterModel CreateParameterModel(ParameterInfo parameterInf
                 throw new ArgumentNullException(nameof(parameterInfo));
             }
 
-            // CoreCLR returns IEnumerable<Attribute> from GetCustomAttributes - the OfType<object>
-            // is needed to so that the result of ToArray() is object
             var attributes = parameterInfo.GetCustomAttributes(inherit: true);
-            var parameterModel = new ParameterModel(parameterInfo, attributes);
 
-            var bindingInfo = BindingInfo.GetBindingInfo(attributes);
-            if (bindingInfo != null)
+            BindingInfo bindingInfo;
+            if (_mvcOptions.AllowValidatingTopLevelNodes && _modelMetadataProvider is ModelMetadataProvider modelMetadataProviderBase)
```

Comment:
Do we need to rename the property to be more expansive in this case? It also means don't use binding info from model metadata now

Reply:
We could change it in rc if we get a lot of feedback about this.